### PR TITLE
mini bootcamp

### PR DESCRIPTION
**Pull request by LakshitXD**

### CODE REVIEW

File: devsnest/db/seeds.rb [hunk 1]
```diff
@@ -16,6 +16,26 @@
...
+Minibootcamp.create(unique_id:"css-mod-2", parent_id: 'CSS', content_type: 2, markdown: 'HELLO CSS')
+Minibootcamp.create(unique_id:"css-mod-3", parent_id: 'CSS', content_type: 2, markdown: 'world')
+Minibootcamp.create(unique_id:"css-mod-4", parent_id: 'CSS', content_type: 2, markdown: 'MODULE 4')
+Minibootcamp.create(unique_id:"css-mod-5", parent_id: 'CSS', content_type: 2, markdown: 'MODULE 5')
+Minibootcamp.create(unique_id:"Javascript", parent_id: 'bootcamp', content_type: 1)
+Minibootcamp.create(unique_id:"js-mod-1", parent_id: 'Javascript', content_type: 2, markdown: 'hello')
+Minibootcamp.create(unique_id:"js-mod-2", parent_id: 'Javascript', content_type: 2, markdown: 'WORLD')
+Minibootcamp.create(unique_id:"js-mod-2", parent_id: 'Javascript', content_type: 2, markdown: 'HELLO JS')
+Minibootcamp.create(unique_id:"js-mod-3", parent_id: 'Javascript', content_type: 2, markdown: 'MODULE 3')
+Minibootcamp.create(unique_id:"js-mod-4", parent_id: 'Javascript', content_type: 2, markdown: 'MODULE 4')
```
Comment (Yug-gurnani) [inline devsnest/db/seeds.rb:41]: Make a separate seed file for Minibootcamp

**Review by Yug-gurnani — CHANGES_REQUESTED** [review]
Make a separate seed file for mini boot camp and fix code climate issues.

File: devsnest/app/controllers/api/v1/minibootcamp_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class MinibootcampController < ApplicationController
+      include JSONAPI::ActsAsResourceController
+    end
+  end
+end
+
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/minibootcamp_controller.rb:10]: remove space

File: devsnest/app/resources/api/v1/frontend_question_resource.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class FrontendQuestionResource < JSONAPI::Resource
+      attributes :module_id, :question_link
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/frontend_question_resource.rb:6]: change it to :question_markdown

File: devsnest/db/migrate/20211208140925_create_minibootcamp.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      t.integer :template
+      t.string :active_path
+      t.text :open_paths
+      t.json :files
+      t.text :protected_paths
+      t.string :video_link
+      t.string :image_url
+      t.boolean :show_explorer
+      t.boolean :show_ide
+      t.timestamps
```
Comment (vedansh) [inline devsnest/db/migrate/20211208140925_create_minibootcamp.rb:18]: should also have a question id

File: devsnest/app/resources/api/v1/minibootcamp_resource.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+module Api
+  module V1
+    # Minibootcamp Resource
+    class MinibootcampResource < JSONAPI::Resource
+      attributes :unique_id, :parent_id, :name, :content_type, :markdown, :video_link, :current_lesson_number, :image_url
+      attributes :module_tasks, :current_url, :next_url, :previous_url, :total_lessons
+      filter :parent_id
+      filter :unique_id
+      filter :content_type
+
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/minibootcamp_resource.rb:12]: sort by priority

Comment (vedansh) [reply devsnest/app/resources/api/v1/minibootcamp_resource.rb:12]: or by lesson number

**Review by Yug-gurnani — APPROVED** [review]

**Pull request merged**